common/Reducers.jsm

```diff
@@ -484,6 +484,33 @@ function DiscoveryStream(prevState = INITIAL_STATE.DiscoveryStream, action) {
           spocs_endpoint: action.data || INITIAL_STATE.DiscoveryStream.spocs.spocs_endpoint,
         },
       };
+    case at.PLACES_LINK_BLOCKED:
+      // Return if action data is empty, or spocs or feeds data is not loaded
+      if (!action.data || !prevState.spocs.loaded || !prevState.feeds.loaded) {
+        return prevState;
+      }
+
+      return {
+        ...prevState,
+        spocs: {
+          ...prevState.spocs,
+          data: prevState.spocs.data.spocs ? {
+            spocs: prevState.spocs.data.spocs.filter(s => s.url !== action.data.url),
+          } : {},
+        },
+        feeds: {
+          ...prevState.feeds,
+          data: Object.keys(prevState.feeds.data).reduce((accumulator, feed_url) => {
+            accumulator[feed_url] = {
+              data: {
+                ...prevState.feeds.data[feed_url].data,
+                recommendations: prevState.feeds.data[feed_url].data.recommendations.filter(r => r.url !== action.data.url),
+              },
+            };
+            return accumulator;
+          }, {}),
```

Comment:
Does blocking an item cause the spocs to be recalculated in `selectLayoutRender.js` and if so, is that the expected behaviour? In other words should this count as a "refresh" the data?

If the expectation is that we *don't* cause a recalculation, you could consider using a flag that tracks whether any changes were made during `filter`.

Reply:
As discussed, any change in state including adding filter flag will cause selectLayoutRender to run. Considering the ask is to remove dismissed item from all components rendered on new tab, it makes sense to update the state but we should fix the [issue raised here](https://github.com/mozilla/activity-stream/pull/4837#pullrequestreview-214090555) in a separate bug